dts/bindings/gpio/microchip,mcp23xxx.yaml

```diff
@@ -31,6 +31,12 @@ properties:
     description: |
       Number of gpios supported by the chip.
 
+  open-drain:
+    type: boolean
+    description: |
+      Specifies that the chip has open-drain rather than push-pull outputs. Should be set
+      for mcp23x09 and mcp23x18 chips.
```

Comment:
Sounds like we ought to have separate bindings for these chips instead?

Reply:
Maybe, but I haven't found a nice way to do that without needlessly copying a bunch of code/configuration when the only difference is really just a single bit.